src/components/form/HdInputPhone.vue

```diff
@@ -0,0 +1,368 @@
+      },
+      smallArrowIcon,
+    };
+  },
+  computed: {
+    sortedCountriesList() {
+      // Preferred countries first
+      if (this.preferredCountries.length) {
+        return COUNTRY_PHONE_CODES.sort((a, b) => {
+          if (this.preferredCountries.includes(a.code) && this.preferredCountries.includes(b.code)) {
+            return this.t.COUNTRIES[a.code] > this.t.COUNTRIES[b.code];
+          }
+          if (this.preferredCountries.includes(a.code)) {
+            return -1;
+          } if (this.preferredCountries.includes(b.code)) {
+            return 1;
+          } if (!this.preferredCountries.includes(a.code) && !this.preferredCountries.includes(b.code)) {
+            return (this.t.COUNTRIES[a.code] > this.t.COUNTRIES[b.code]) ? 1 : -1;
+          }
+          return 0;
+        });
+      }
+      return COUNTRY_PHONE_CODES;
+    },
+    arrowClassNames() {
+      const baseClass = 'phone-input__selector__arrow';
+      const directionClass = this.showDropdown ? 'phone-input__selector__arrow--down' : 'phone-input__selector__arrow--up';
+      return [baseClass, directionClass];
+    },
+    t() {
+      return getMessages(this.lang);
+    },
+  },
+  watch: {
+    phoneNumber(newPhoneNumber) {
+      const newSelectedCountry = COUNTRY_PHONE_CODES.find((country) => (
+        country.dial_code === newPhoneNumber.substring(0, MIN_DIGITS_DIAL_CODE)
+        || country.dial_code === newPhoneNumber.substring(0, MIN_DIGITS_DIAL_CODE + 1)
+        || country.dial_code === newPhoneNumber.substring(0, MAX_DIGITS_DIAL_CODE)
+      ));
+      if (newSelectedCountry) this.selectedCountry = newSelectedCountry;
+      action('input')(newPhoneNumber);
+    },
+    selectedCountry(newCountry, prevCountry) {
+      if (this.phoneNumber.length >= MAX_DIGITS_DIAL_CODE) {
+        this.phoneNumber = this.phoneNumber.replace(prevCountry.dial_code, newCountry.dial_code);
+      } else {
+        this.phoneNumber = newCountry.dial_code;
+      }
+    },
+    showDropdown(isShown) {
+      let selectKeys;
+      if (isShown) {
+        selectKeys = window.addEventListener('keyup', this.keysHandler);
+      } else {
+        window.removeEventListener('keyup', selectKeys);
+      }
+    },
+  },
+  methods: {
+    toggleDropdown() {
+      this.showDropdown = !this.showDropdown;
+    },
+    selectCountry(country) {
+      this.selectedCountry = country;
+      this.toggleDropdown();
+      if (this.phoneNumber.length <= MAX_DIGITS_DIAL_CODE) {
+        this.$refs.input.$refs.input.$refs.input.focus();
+      }
+    },
+    phoneFormatter(phone) {
+      return formatPhoneNumber(this.selectedCountry.dial_code, phone);
+    },
+    keysHandler(event) {
+      const { keyCode } = event;
+      const firstCountryCode = this.sortedCountriesList[0].code;
+      const lastCountryCode = this.sortedCountriesList[this.sortedCountriesList.length - 1].code;
+      let countryNode = null;
+
+      const foucsCountryCode = () => {
```

Comment:
`foucsCountryCode` => `focusCountryCode` 🤓  - I am using cSpell on my vscode. I will share the URL with you in case if you want to see it as well: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

Reply:
Yes! Thank you 🙏🏼